inc/saipoe.h

```diff
@@ -0,0 +1,522 @@
+     * @objects SAI_OBJECT_TYPE_POE_PORT
+     */
+    SAI_POE_DEVICE_ATTR_POE_PORT_LIST,
+
+    /**
+     * @brief The total power in the device, in watts
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_POE_DEVICE_ATTR_TOTAL_POWER,
+
+    /**
+     * @brief Total power consumption, in MILLI watts
+     *
+     * @type sai_uint32_t
+     * @flags READ_ONLY
+     */
+    SAI_POE_DEVICE_ATTR_POWER_CONSUMPTION,
+
+    /**
+     * @brief POE device version and information (POE firmware version)
+     *
+     * @type char
+     * @flags READ_ONLY
+     */
+    SAI_POE_DEVICE_ATTR_VERSION,
+
+    /**
+     * @brief Power limit mode
+     *
+     * @type sai_poe_device_limit_mode_t
+     * @flags CREATE_AND_SET
+     * @default SAI_POE_DEVICE_LIMIT_MODE_CLASS
+     */
+    SAI_POE_DEVICE_ATTR_POWER_LIMIT_MODE,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_POE_DEVICE_ATTR_END,
+
+    /** Custom range base value */
+    SAI_POE_DEVICE_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_POE_DEVICE_ATTR_CUSTOM_RANGE_END
+
+} sai_poe_device_attr_t;
+
+/**
+ * @brief POE PSE attributes
+ */
+typedef enum _sai_poe_pse_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_POE_PSE_ATTR_START,
+
+    /**
+     * @brief POE PSE ID (index)
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_POE_PSE_ATTR_ID = SAI_POE_PSE_ATTR_START,
+
+    /**
+     * @brief POE device ID
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_POE_DEVICE
+     */
+    SAI_POE_PSE_ATTR_DEVICE_ID,
+
+    /**
+     * @brief POE PSE PSE software version
+     *
+     * @type char
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_PSE_SOFTWARE_VERSION,
+
+    /**
+     * @brief POE PSE PSE hardware version
+     *
+     * @type char
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_PSE_HARDWARE_VERSION,
+
+    /**
+     * @brief Temperature (in Celsius) of the PSE
+     *
+     * @type sai_int32_t
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_TEMPERATURE,
+
+    /**
+     * @brief Status of the PSE
+     *
+     * @type sai_poe_pse_status_t
+     * @flags READ_ONLY
+     */
+    SAI_POE_PSE_ATTR_PSE_STATUS,
+
+    /**
+     * @brief End of attributes
+     */
+    SAI_POE_PSE_ATTR_END,
+
+    /** Custom range base value */
+    SAI_POE_PSE_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_POE_PSE_ATTR_CUSTOM_RANGE_END
+
+} sai_poe_pse_attr_t;
+
+/**
+ * @brief POE port attributes
+ */
+typedef enum _sai_poe_port_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_POE_PORT_ATTR_START,
+
+    /**
+     * @brief POE port front panel ID
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_POE_PORT_ATTR_FRONT_PANEL_ID = SAI_POE_PORT_ATTR_START,
```

Comment:
Is it the same as `SAI_PORT_ATTR_HW_LANE_LIST` ? Is so, should we add `| KEY` as well?
```
     * @type sai_u32_list_t
     * @flags MANDATORY_ON_CREATE | CREATE_ONLY | KEY
     */
    SAI_PORT_ATTR_HW_LANE_LIST,
```
Can we have a single POE port that uses multiple HW lanes?

Reply:
The above diagram has a few glitches. The PSE ports are p1 ... pN. Here the N could be 4 or 8. This is subject to the PoE chip and the supported PoE mode of operations like AT, BT-Type3, BT-full (Both Type-3, Type-4).
Similarly the Host CPU to PoE MCU interface could be Shared Memory/UART/I2C/SPI etc. It is subject to the platform and the implementation.

Reply:
In general, the image wasn't generic enough. It was updated in PoE_Physical_Design.png to reflect something less constrictive.